core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/RequestBody.java

```diff
@@ -169,6 +169,18 @@ public static RequestBody fromByteBuffer(ByteBuffer byteBuffer) {
         return fromBytesDirect(BinaryUtils.copyAllBytesFrom(byteBuffer));
     }
 
+    /**
+     * Creates a {@link RequestBody} from the remaining readable bytes from a {@link ByteBuffer}. Unlike
+     * {@link #fromByteBuffer(ByteBuffer)}, this method respects the current read position of the buffer and reads only
+     * the remaining bytes. The buffer is copied before reading so no changes are made to original buffer.
+     *
+     * @param byteBuffer ByteBuffer to send to the service.
+     * @return RequestBody instance.
+     */
```

Comment:
Can we update the javadoc in the original fromByteBuffer(...) to make it clear that this one does -not- respect the current read position.

Reply:
+1